diff --git a/Makefile.arm b/Makefile.arm
index 62bf275..f4f509f 100644
--- a/Makefile.arm
+++ b/Makefile.arm
@@ -1,27 +1,27 @@
 # ifeq logical or
 ifeq ($(CORE), $(filter $(CORE),CORTEXA9 CORTEXA15))
 ifeq ($(OSNAME), Android)
-CCOMMON_OPT += -marm -mfpu=neon  -mfloat-abi=hard -march=armv7-a
-FCOMMON_OPT += -marm -mfpu=neon  -mfloat-abi=hard -march=armv7-a
+CCOMMON_OPT += -marm -mfpu=neon  -mfloat-abi=softfp -march=armv7-a
+FCOMMON_OPT += -marm -mfpu=neon  -mfloat-abi=softfp -march=armv7-a
 else
-CCOMMON_OPT += -marm -mfpu=vfpv3  -mfloat-abi=hard -march=armv7-a
-FCOMMON_OPT += -marm -mfpu=vfpv3  -mfloat-abi=hard -march=armv7-a
+CCOMMON_OPT += -marm -mfpu=vfpv3  -mfloat-abi=softfp -march=armv7-a
+FCOMMON_OPT += -marm -mfpu=vfpv3  -mfloat-abi=softfp -march=armv7-a
 endif
 endif
 
 ifeq ($(CORE), ARMV7)
 ifeq ($(OSNAME), Android)
-CCOMMON_OPT += -marm -mfpu=neon  -mfloat-abi=hard -march=armv7-a -Wl,--no-warn-mismatch
-FCOMMON_OPT += -marm -mfpu=neon  -mfloat-abi=hard -march=armv7-a -Wl,--no-warn-mismatch
+CCOMMON_OPT += -marm -mfpu=neon  -mfloat-abi=softfp -march=armv7-a -Wl,--no-warn-mismatch
+FCOMMON_OPT += -marm -mfpu=neon  -mfloat-abi=softfp -march=armv7-a -Wl,--no-warn-mismatch
 else
-CCOMMON_OPT += -marm -mfpu=vfpv3  -mfloat-abi=hard -march=armv7-a
-FCOMMON_OPT += -marm -mfpu=vfpv3  -mfloat-abi=hard -march=armv7-a
+CCOMMON_OPT += -marm -mfpu=vfpv3  -mfloat-abi=softfp -march=armv7-a
+FCOMMON_OPT += -marm -mfpu=vfpv3  -mfloat-abi=softfp -march=armv7-a
 endif
 endif
 
 ifeq ($(CORE), ARMV6)
-CCOMMON_OPT += -marm -mfpu=vfp -mfloat-abi=hard  -march=armv6
-FCOMMON_OPT += -marm -mfpu=vfp -mfloat-abi=hard  -march=armv6
+CCOMMON_OPT += -marm -mfpu=vfp -mfloat-abi=softfp  -march=armv6
+FCOMMON_OPT += -marm -mfpu=vfp -mfloat-abi=softfp  -march=armv6
 endif
 
 
